### DISCUSSION TITLE
useRef doesn't work with elements inside Drawer/Modal

### DISCUSSION THREAD

**Discussion by empz** [mantinedev/mantine#6464] [Q&A] [answered]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.10.2

### What package has an issue?

@mantine/core

### What framework do you use?

Next.js

### In which browsers you can reproduce the issue?

All

### Describe the bug

`useRef` doesn't work with elements inside a `<Drawer>` and `<Modal>` (and probably any other Portal-based component).

### If possible, include a link to a codesandbox with a minimal reproduction

https://codesandbox.io/p/devbox/mantine-drawer-useref-bug-f2szsh

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by rtivital** [comment, marked as answer]
You can get ref only of the element that is preset in the DOM. Content of Drawer/Modal is unmounted when Drawer/Modal is not opened. It is intended behavior. If you want to keep elements mounted, set `keepMounted` prop.
